diff -uarpN alitengine/auto/cc/conf alitengine_linker/auto/cc/conf
--- alitengine/auto/cc/conf	2015-02-04 19:46:59.636174967 +0800
+++ alitengine_linker/auto/cc/conf	2015-02-04 20:01:12.556149716 +0800
@@ -3,7 +3,7 @@
 # Copyright (C) Nginx, Inc.
 
 
-LINK="\$(CC)"
+LINK=${LINK:-"\$(CC)"}
 
 ngx_include_opt="-I "
 ngx_compile_opt="-c"
diff -uarpN alitengine/auto/options alitengine_linker/auto/options
--- alitengine/auto/options	2015-02-04 19:46:59.668174966 +0800
+++ alitengine_linker/auto/options	2015-02-04 20:02:31.524147378 +0800
@@ -663,6 +663,7 @@ use the \"--without-http_limit_conn_modu
 
         --with-cc=*)                     CC="$value"                ;;
         --with-cpp=*)                    CPP="$value"               ;;
+        --with-link=*)                   LINK="$value"              ;;
         --with-cc-opt=*)                 NGX_CC_OPT="$value"        ;;
         --with-ld-opt=*)                 NGX_LD_OPT="$value"        ;;
         --with-cpu-opt=*)                CPU="$value"               ;;
@@ -920,6 +921,7 @@ cat << END
 
   --with-cc=PATH                     set C compiler pathname
   --with-cpp=PATH                    set C preprocessor pathname
+  --with-link=LINK                   set C linker
   --with-cc-opt=OPTIONS              set additional C compiler options
   --with-ld-opt=OPTIONS              set additional linker options
   --with-cpu-opt=CPU                 build for the specified CPU, valid values:
